File: python/0122_best_time_to_buy_and_sell_stock_II.py
# n = len(prices)
# Time: O(n)
# Space: O(1)
from typing import List


class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0

        for idx in range(1, len(prices)):
            if prices[idx] - prices[idx - 1] > 0:
                profit += prices[idx] - prices[idx - 1]

        return profit

File: go/0122_best_time_to_buy_and_sell_stock_II.go
package main

// n = len(prices)
// Time: O(n)
// Space: O(1)
func maxProfit(prices []int) int {
	profit := 0

	for i := 0; i < len(prices)-1; i++ {
		if prices[i+1]-prices[i] > 0 {
			profit += prices[i+1] - prices[i]
		}
	}

	return profit
}
